CMakeLists.txt

```diff
@@ -28,14 +29,18 @@ catkin_package(
   DEPENDS
 )
 
+catkin_add_gtest(test_buffer test/TestBuffer.cpp)
+target_link_libraries(test_buffer ${GTEST_BOTH_LIBRARIES} pthread gmock ${Eigen3_LIBRARIES})
+target_link_libraries(test_buffer ${PROJECT_NAME} ${catkin_LIBRARIES})
+include_directories(${GTEST_INCLUDE_DIRS} gmock pthread)
 
 ###########
 ## Build ##
 ###########
 
 include_directories(${catkin_INCLUDE_DIRS} 
                     ${EIGEN3_INCLUDE_DIRS}
-					include)
+					include/${PROJECT_NAME}/)
```

Comment:
include/${PROJECT_NAME}/ cannot find the proper include directory.

The correct way is:
include_directories(${catkin_INCLUDE_DIRS}  ${EIGEN3_INCLUDE_DIRS}
					include)